lib/puppet/trusted_external.rb

```diff
@@ -3,11 +3,35 @@ module Puppet::TrustedExternal
   def retrieve(certname)
     command = Puppet[:trusted_external_command]
     return nil unless command
+    unless Puppet::FileSystem.directory?(command)
+      return fetch_data(command, certname)
+    end
+    # command is a directory. Thus, data is a hash of <basename> => <data> for
+    # each executable file in command. For example, if the files 'servicenow.rb',
+    # 'unicorn.sh' are in command, then data is the following hash:
+    #   { 'servicenow' => <servicenow.rb output>, 'unicorn' => <unicorn.sh output> }
+    data = {}
+    Puppet::FileSystem.children(command).each do |file|
+      executable_file = Puppet::FileSystem.file?(file) && Puppet::FileSystem.executable?(file)
+      next unless executable_file
+      basename = File.basename(file, file.extname)
+      unless data[basename].nil?
+        # TODO: Can happen if we have servicenow.rb, servicenow.sh in command. Should
+        # we raise an exception here? If yes, how fine-grained should exception be?
```

Comment:
I'm thinking yes, even if it is an edge-case. Re: the fine-grained bits, I'm wondering if we should have an exception that is `"You have conflicting scripts and here they are: <scripts>"` or instead is `"There is more than one <first_conflicting_basename> script in #{external_command_dir"`

Reply:
I agree—we should raise an exception. We don't need to get more granular than your suggestion. It wouldn't hurt to enumerate them if it's easy but specifying the basename with the conflict is decent.

Reply:
Updated to go with "list just the first one"